### PR TITLE
🐛 Only propagate the deletionTimestamp to a location if there is a syncer finalizer for it.

### PR DESCRIPTION
**Pull request by jmprusi**
## Summary

The deletionTimestamp propagation should only happen if there is a syncer finalizer holding that location.

This PR fixes a situation where the syncer and the resource controller will enter a reconciliation storm, due to the resource controller propagating the deletion timestamp of a resource, and the syncer removing its location deletionTimestamp as it has been removed from the downstream cluster.

## Related issue(s)
Fixes https://github.com/kcp-dev/kcp/issues/1891

### CODE REVIEW

File: pkg/reconciler/workload/resource/resource_reconcile.go [hunk 1]
```diff
@@ -151,15 +151,29 @@ func (c *Controller) reconcileResource(ctx context.Context, lclusterName logical
...
+		// If the object doesn't have a syncer finalizer for the given location, we don't need to set the deletion timestamp.
+		if !hasFinalizerForLocation {
+			break
+		}
 		if val, ok := objAnnotations[workloadv1alpha1.InternalClusterDeletionTimestampAnnotationPrefix+location]; !ok || val == "" {
 			annotationPatch[workloadv1alpha1.InternalClusterDeletionTimestampAnnotationPrefix+location] = obj.GetDeletionTimestamp().Format(time.RFC3339)
 		}
 	}
+
+	logger.Info("resource", "annotations", obj.GetAnnotations(), "finalizers", obj.GetFinalizers(), "labels", obj.GetLabels(), "patch", annotationPatch)
```
Comment (stevekuznetsov) [inline pkg/reconciler/workload/resource/resource_reconcile.go:176]: Logging all annotations & labels is going to be pretty verbose. Can we extract out the important bits?

Comment (jmprusi) [reply pkg/reconciler/workload/resource/resource_reconcile.go:176]: let's remove it,just added it for debugging.

**Review by stevekuznetsov — APPROVED** [review]
/hold

This seems reasonable. Deferring to @davidfestal for merge

**Comment by davidfestal** [comment]
Isn't there a risk that by fixing the problem this way on the resource controller side, we may loose the "intent to remove from a location" information ?
The fact that no syncer finalizer exist for a given SynTarget only means that the corresponding Syncer still didn't take ownership of the resource at this time, but it might be that the Syncer is previsl in the process of adding this finalizer.

I'm wondering whether the fix should rather be done on the Syncer side, to change how we react to the presence of the deletion annotation.

**Comment by davidfestal** [comment]
After discussing about it, it seems that a more proper fix in the resource controller would be, at least as a first step, to avoid scheduling a KCP resource at all (== setting the `RequestState` label) if it has a deletion timestamp.

**Comment by jmprusi** [comment]
Closed in favor of https://github.com/kcp-dev/kcp/pull/1953 as not propagating the deletionTimestamp can have some unintended consequences.

**Pull request closed without merging**